graph/db/graph_test.go

```diff
@@ -4063,3 +4063,83 @@ func TestClosedScid(t *testing.T) {
 	require.Nil(t, err)
 	require.True(t, exists)
 }
+
+// TestUpdateAliasChannel tests that the underlying graph channel information
+// for an alias channel is deleted and readded under a different short channel
+// id.
+func TestUpdateAliasChannel(t *testing.T) {
+	t.Parallel()
+
+	graph, err := MakeTestGraph(t)
+	require.NoError(t, err, "unable to make test database")
+
+	// Create first node and add it to the graph.
+	node1, err := createTestVertex(graph.db)
+	require.NoError(t, err, "unable to create test node")
+	err = graph.AddLightningNode(node1)
+	require.NoError(t, err)
+
+	// Create second node and add it to the graph.
+	node2, err := createTestVertex(graph.db)
+	require.NoError(t, err, "unable to create test node")
+	err = graph.AddLightningNode(node2)
+	require.NoError(t, err)
+
+	// Adding a new channel edge to the graph.
+	edgeInfo, edge1, edge2 := createChannelEdge(
+		graph.db, node1, node2,
+	)
+	if err := graph.AddChannelEdge(edgeInfo); err != nil {
+		t.Fatalf("unable to create channel edge: %v", err)
+	}
+
+	// Add both edge policies.
+	err = graph.UpdateEdgePolicy(edge1)
+	require.NoError(t, err)
+
+	err = graph.UpdateEdgePolicy(edge2)
+	require.NoError(t, err)
+
+	oldSCID := edgeInfo.ChannelID
+
+	// Define the new channel id (under real conditions this is the new
+	// confirmed channel id).
+	newChanSCID := rand.Uint64()
```

Comment:
But made the newChanID deterministic:
```
newChanSCID := oldSCID + 1
```

Reply:
> no it should succeed aslo with the same SCID, in case of an option-scid channel but not zeroconf.

I propose to unit test that when the same SCID is passed, ReAddChannelEdge succeeds in case of an option-scid channel, but fails for a zeroconf channel.

Reply:
hmm I think I was not clear in my description what I meant is that for zeroconf-alias channels the channelid for the reassignment is always different,
For non-zeroconf alias channels the ChannelID for the reassignment is always the same, we do this only to clear the graph data storage to remove potential ChanUpdates which where based on the alias not the channelID. Which might happen in the period between channelconfirmation (usually 3 blocks) and the 6 blocks public gossip limit.